src/rust/engine/src/nodes.rs

```diff
@@ -405,7 +398,7 @@ impl ExecuteProcess {
       level,
       append_only_caches,
       jdk_home,
-      platform_constraint,
+      platform_constraint: Some(process_config.platform),
```

Comment:
We could simplify this to no longer be an `Option`, but I figured maybe it's easier to keep it in place for part 2 of https://github.com/pantsbuild/pants/issues/16873?